swank.asd

```diff
@@ -9,17 +9,28 @@
 
 ;; Usage:
 ;;
-;;   (asdf:load-system "swank")
-;;   (swank:create-swank-server PORT) => ACTUAL-PORT
+;;   (asdf:load-system :swank)
+;;   (swank:create-server :dont-close t)
 ;;
-;; (PORT can be zero to mean "any available port".)
-;; Then the Swank server is running on localhost:ACTUAL-PORT. You can
+;; After which, the Swank server is running on localhost:4005. You can
 ;; use `M-x slime-connect' to connect Emacs to it.
 ;;
 ;; This code has been placed in the Public Domain.  All warranties
 ;; are disclaimed.
 
 (asdf:defsystem "swank"
+  :perform (load-op :after (o c)
+              (set (intern '*source-directory* 'swank-loader)
+                   (asdf:system-source-directory :swank))
+              (set (intern '*fasl-directory* 'swank-loader)
+                   (asdf:apply-output-translations (asdf:system-source-directory :swank)))
```

Comment:
these calls to intern break the build, because the argument should be a string (like in calls below(!)). After fixing that all works fine with ECL.

```diff
modified   swank.asd
@@ -20,9 +20,9 @@
 
 (asdf:defsystem "swank"
   :perform (load-op :after (o c)
-              (set (intern '*source-directory* 'swank-loader)
+              (set (intern "*SOURCE-DIRECTORY*" 'swank-loader)
                    (asdf:system-source-directory :swank))
-              (set (intern '*fasl-directory* 'swank-loader)
+              (set (intern "*FASL-DIRECTORY*" 'swank-loader)
                    (asdf:apply-output-translations (asdf:system-source-directory :swank)))
               (uiop:symbol-call :swank :before-init
                  (uiop:symbol-call :swank-loader :slime-version-string)
```

Reply:
Thanks for the review!